source/MaterialXGenShader/ShaderNode.cpp

```diff
@@ -206,10 +206,24 @@ ShaderNodePtr ShaderNode::create(const string& name, const NodeDef& nodeDef, Sha
         }
         else
         {
-            ShaderInput* input = newNode->addInput(elem->getName(), TypeDesc::get(elem->getType()));
-            if (!elem->getValueString().empty())
+            ShaderInput* input = nullptr;
```

Comment:
Check for type of implementation and set value based on nodedef. This is where we switch types from string to int based on implementation.